src/fidesops/service/connectors/sql_connector.py

```diff
@@ -187,6 +196,58 @@ def create_client(self) -> Engine:
             echo=not self.hide_parameters,
         )
 
+    def set_schema(self, connection: Connection) -> None:
+        """Sets the search_path for the duration of the session"""
+        config = RedshiftSchema(**self.configuration.secrets or {})
+        if config.db_schema:
+            logger.info("Setting Redshift search_path before retrieving data")
+            connection.execute("SET search_path TO %s" % config.db_schema)
+
+    # Overrides SQLConnector.retrieve_data
+    def retrieve_data(
+        self, node: TraversalNode, policy: Policy, input_data: Dict[str, List[Any]]
+    ) -> List[Row]:
+        """Retrieve data from Amazon Redshift
+
+        For redshift, we also set the search_path to be the schema defined on the ConnectionConfig if
+        applicable - persists for the current session.
+        """
+        query_config = self.query_config(node)
+        client = self.client()
+        stmt = query_config.generate_query(input_data, policy)
+        if stmt is None:
+            return []
+
+        logger.info(f"Starting data retrieval for {node.address}")
+        with client.connect() as connection:
+            self.set_schema(connection)
```

Comment:
If this persists for the duration of the session, do we need to call it every time we retrieve data?

Reply:
Maybe i phrased it wrong, it lasts until this connection is closed -

Reply:
I originally tried to set it more broadly, but then the schema doesn't seem to be set for individual connections -